tests/unit/Files/DTO/FileTest.php

```
@@ -0,0 +1,270 @@
+    }
+
+    /**
+     * Tests creating a File from a URL with inferred MIME type.
+     *
+     * @return void
+     */
+    public function testCreateFromUrlWithInferredMimeType(): void
+    {
+        $url = 'https://example.com/document.pdf';
+        
+        $file = new File($url);
+        
+        $this->assertEquals(FileTypeEnum::remote(), $file->getFileType());
+        $this->assertEquals($url, $file->getUrl());
+        $this->assertEquals('application/pdf', $file->getMimeType());
+        $this->assertFalse($file->isText());
+    }
+
+    /**
+     * Tests creating a File from a data URI.
+     *
+     * @return void
+     */
+    public function testCreateFromDataUri(): void
+    {
+        $base64Data = 'SGVsbG8gV29ybGQ=';
+        $dataUri = 'data:text/plain;base64,' . $base64Data;
+        
+        $file = new File($dataUri);
+        
+        $this->assertEquals(FileTypeEnum::inline(), $file->getFileType());
+        $this->assertNull($file->getUrl());
+        $this->assertEquals($base64Data, $file->getBase64Data());
+        $this->assertEquals($dataUri, $file->getDataUri());
+        $this->assertEquals('text/plain', $file->getMimeType());
+        $this->assertTrue($file->isText());
+    }
+
+    /**
+     * Tests creating a File from a data URI with provided MIME type override.
+     *
+     * @return void
+     */
+    public function testCreateFromDataUriWithMimeTypeOverride(): void
+    {
+        $base64Data = 'SGVsbG8gV29ybGQ=';
+        $dataUri = 'data:text/plain;base64,' . $base64Data;
+        $overrideMimeType = 'text/html';
+        
+        $file = new File($dataUri, $overrideMimeType);
+        
+        $this->assertEquals(FileTypeEnum::inline(), $file->getFileType());
+        $this->assertEquals($base64Data, $file->getBase64Data());
+        $this->assertEquals($overrideMimeType, $file->getMimeType());
+        $this->assertEquals('data:text/html;base64,' . $base64Data, $file->getDataUri());
+    }
```

Seeing this now, I wonder whether we should even allow for such an inconsistency. I think it's okay to consider it as an "override MIME type", but it could also be considered an accident that should throw an exception, when the MIME types differ.

---

Feels like a niche situation. I suspect folks will generally try without providing the MIME type, only providing it if they're giving a Base64 or something that requires it. To take your point from earlier, this feels like the kind of situation where our validation could be interpreted as wrong.